src/constructs/autoscaling/asg.test.ts

```diff
@@ -295,4 +296,85 @@ describe("The GuAutoScalingGroup", () => {
       },
     });
   });
+
+  test("has an instance role created by default with AssumeRole permissions", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
+      app: "TestApp",
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+      },
+    });
+  });
+
+  test("passing in an instance role overrides the default", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      ...defaultProps,
+      app: "TestApp",
+      role: new GuInstanceRole(stack, {
+        app: "OverridenAppName",
+      }),
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      Tags: [
+        { Key: "App", Value: "OverridenAppName" },
+        { Key: "gu:cdk:version", Value: "TEST" },
+        { Key: "Stack", Value: "test-stack" },
+        { Key: "Stage", Value: { Ref: "Stage" } },
+      ],
```

Comment:
There are a [couple of helpers](https://github.com/guardian/cdk/blob/b7b79365b5218b15d3bb798ac46e85b092af90a1/src/constructs/core/stack.test.ts#L41-L56) we can use for testing tags that we can use here.

Reply:
Thanks, implemented in a new push.